compiam/dunya/__init__.py

```diff
@@ -77,14 +88,53 @@ def get_recording(self, rmbid):
         """
         return _dunya_query_json("api/" + self.tradition + "/recording/%s" % rmbid)
 
-    def list_artists(self):
-        """List the artists in the database. This function will automatically page through API results.
+    def _get_metadata(self):
+        """Query a list of unique identifiers per each relevant tag in the Dunya database. This 
+        method is automatically run when the corpora is initialized.
 
-        :returns: A list of dictionaries containing artist information:
-            ``{"mbid": MusicBrainz artist id, "name": Name of the artist}``.
-            For additional information about each artist use :func:`get_artist`.
+        :returns: A dictionary of lists of unique identifiers per each tag: artists, concerts, works,
+        raagas, taalas, and instruments. It also includes a complete list of recording ids for the collection
         """
-        return _get_paged_json("api/" + self.tradition + "/artist")
+        recording_list = []
+        artist_list = []
+        concert_list = []
+        work_list = []
+        raga_list = []
+        tala_list = []
+        instrument_list = []
+
+        for rec in tqdm.tqdm(self.get_collection(), desc="Parsing metadata from database"):
+            try:
+                # Getting artist list
+                for artist in self.get_recording(rec["mbid"])["artists"]:
+                    if artist["lead"]:
+                        artist_list.append(artist["artist"])
+                    instrument_list.append(artist["instrument"])
+                # Getting concert list
+                for concert in self.get_recording(rec["mbid"])["concert"]:
+                    concert_list.append(concert)
+                # Getting work list
+                for work in self.get_recording(rec["mbid"])["work"]:
+                    work_list.append(work)
+                # Getting raaga list
+                for raga in self.get_recording(rec["mbid"])["raaga"]:
+                    raga_list.append(raga)
+                # Getting taala list
+                for tala in self.get_recording(rec["mbid"])["taala"]:
+                    tala_list.append(tala)
+                recording_list.append(rec["mbid"])
+            except:
+                continue
```

Comment:
Could probably be more explicit here about when you are happy for it to fail.

Reply:
hhhmmm OK, but I would not want the code to stop. There are cases where the track is listed in the database but the audio/features are not actually available, which makes the code to fail, and we don't want that.

Reply:
When the audio/features are not available what is the error? I guess KeyError? So we can catch that error as except KeyError: and log the error as a warning or leave a comment explaining why we don't mind it failing.

Thjen if there is some other error we do care about it will be raised...

Anyway, not the **most** important thing but maybe a good practice :p

Reply:
I see, that makes sense. On that note, I think we should also take a look at this issue (#82). It's probably related to that and it applied to the entirety of compIAM actually.

Reply:
so maybe we can leave it like this, and improve this matter together with all `prints` and `logs` and `raises` in compIAM. What do you think @thomasgnuttall?